examples/image_ocr.py

```diff
@@ -0,0 +1,433 @@
+        if epoch == 10:
+            self.build_word_list(8, 1)
+        if epoch == 20:
+            self.build_word_list(8, 0.5)
+        if epoch == 30:
+            self.build_word_list(12, 0.5)
+
+# the actual loss calc occurs here despite it not being
+# a true Keras loss function
+def ctc_lambda_func(inputs):
+    # theano doesn't give the last operation a name for some reason,
+    # while tensorflow adds ":0" to names..so use a complicated cleanup func
+    inputs = {str(i.name).split(":")[0]: i for i in inputs}
+    # ..and some package specific code
+    if K._BACKEND == "tensorflow":
+        y_pred = inputs["div"]
+    else:
+        y_pred = inputs["None"]
+    # the 2 is critical here since the first couple outputs of the RNN
+    # tend to be garbage:
+    y_pred = y_pred[:, 2:, :]
+
+    return K.ctc_batch_cost(inputs["the_labels"], y_pred,
+                            inputs["input_length"], inputs["label_length"])
+
+# keras still wants a loss func in this particular parameter format
+def dummy_func(y_true_dummy, y_pred):
+    return y_pred
+
+# For a real OCR application, this should be beam search with a dictionary
+# and language model.  For this example, best path is sufficient.
+def decode_batch(test_func, word_batch):
+    import itertools
+    out = test_func([word_batch])[0]
+    ret = []
+    for j in range(out.shape[0]):
+        out_best = list(np.argmax(out[j, 2:], 1))
+        out_best = [k for k, g in itertools.groupby(out_best)]
+        # 26 is space, 27 is CTC blank char
+        outstr = ""
+        for c in out_best:
+            if c >= 0 and c < 26:
+                outstr += chr(c + ord('a'))
+            elif c == 26:
+                outstr += " "
+        ret.append(outstr)
+    return ret
+
+
+class VizCallback (keras.callbacks.Callback):
+
+    num_display_words = 6
+
+    def __init__(self, test_func, text_img_gen):
+        self.test_func = test_func
+        self.outputDir = datetime.datetime.now().strftime("image_ocr %A, %d. %B %Y %I.%M%p")
+        self.text_img_gen = text_img_gen
+        os.mkdir(self.outputDir)
+
+    def show_edit_distance(self, num):
+        num_left = num
+        mean_norm_ed = 0.0
+        mean_ed = 0.0
+        while(num_left > 0):
+            word_batch = next(self.text_img_gen)[0]
+            num_proc = min(word_batch["the_input"].shape[0], num_left)
+            decoded_res = decode_batch(self.test_func, word_batch["the_input"][0:num_proc])
+            for j in range(0, num_proc):
+                edit_dist = editdistance.eval(decoded_res[j], word_batch["source_str"][j])
+                mean_ed += float(edit_dist)
+                mean_norm_ed += float(edit_dist)/len(word_batch["source_str"][j])
+            num_left -= num_proc
+        mean_norm_ed = mean_norm_ed / num
+        mean_ed = mean_ed / num
+        print "\nOut of %d samples:  Mean edit distance: %.3f Mean word-length normalized edit distance: %0.3f" \
```

Comment:
Print statements should be cross-compatible py2/3. See how it is handled in other examples.